lib/bucket_store/key_storage.rb

```diff
@@ -40,45 +122,32 @@ def filename
     # @example Download a key
     #   BucketStore.for("inmemory://bucket/file.xml").download
     def download
-      raise ArgumentError, "Key cannot be empty" if key.empty?
-
-      BucketStore.logger.info(event: "key_storage.download_started")
-
-      start = BucketStore::Timing.monotonic_now
-      result = adapter.download(bucket: bucket, key: key)
-
-      BucketStore.logger.info(event: "key_storage.download_finished",
-                              duration: BucketStore::Timing.monotonic_now - start)
-
-      result
+      buffer = StringIO.new
+      stream.download(file: buffer).tap do |result|
+        result.delete(:file)
+        result[:content] = buffer.string
+      end
```

Comment:
I think we're changing the logging structure with this approach as everything is now logged under `key_storage.stream.*`. I can live with this but seems like a breaking change? Perhaps we can just remove the `.stream` from `KeyStreamer` though

Reply:
So I defaulted back to not changing the log line e.g. if you're using `streaming` or regular you'll get the same logs. Couldn't figure out a good ROI story for separating them